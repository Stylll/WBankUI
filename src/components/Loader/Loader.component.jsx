import React from 'react';
import PropTypes from 'prop-types';

import './Loader.styles.scss';

const Loader = (props) => {
  const { customStyles, message, containerClassName } = props;

  return (
    <div className={`loader-container ${containerClassName}`}>
      <div className="loader" style={customStyles} />
      {message && <span className="message">{message}</span>}
    </div>
  );
};

Loader.propTypes = {
  customStyles: PropTypes.object,
  containerClassName: PropTypes.string,
  message: PropTypes.string,
};

Loader.defaultProps = {
  customStyles: {},
  containerClassName: '',
};

export default Loader;
