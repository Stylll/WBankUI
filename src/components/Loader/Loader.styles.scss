@import '../../partials/colors';

.loader-container {
  display: flex;
  align-items: center;
  .loader {
    border: 3px solid $primary-color;
    border-top: 3px solid $white;
    border-radius: 50%;
    width: 5px;
    height: 5px;
    animation: spin 1.5s linear infinite;
    margin: 4px;
  }
  .message {
    font-size: 12px;
  }
}


@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}
