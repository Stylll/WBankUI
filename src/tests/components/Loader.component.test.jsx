import { render } from '@testing-library/react';
import React from 'react';

import Loader from '../../components/Loader/Loader.component';

describe('Loader Component', () => {
  it('should render properly', () => {
    const { baseElement } = render(<Loader />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should render properly with message', () => {
    const { baseElement } = render(<Loader message="custom message" />);

    expect(baseElement).toMatchSnapshot();
  });
});
