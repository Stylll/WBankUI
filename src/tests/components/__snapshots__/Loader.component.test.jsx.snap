// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component should render properly 1`] = `
<body>
  <div>
    <div
      class="loader-container "
    >
      <div
        class="loader"
      />
    </div>
  </div>
</body>
`;

exports[`Loader Component should render properly with message 1`] = `
<body>
  <div>
    <div
      class="loader-container "
    >
      <div
        class="loader"
      />
      <span
        class="message"
      >
        custom message
      </span>
    </div>
  </div>
</body>
`;
